### DISCUSSION TITLE
Marvel vs Capcom 2 (Naomi): "Grid" artifacts on AMD APU

### DISCUSSION THREAD

**Discussion by kathyrollo** [flyinghead/flycast#1086] [Q&A] [answered]
Hi @flyinghead ,

I'm having a very similar problem to [this](https://github.com/libretro/flycast/issues/52) issue in the standalone version (v2.1). This wasn't a problem on my NVIDIA GPU as far as I can recall, but I recently got a [Beelink SER5 Pro](https://www.bee-link.com/beelink-amd-ryzen-5-ser5-5800u-minipc-clone-1) (Ryzen 7 5800H) mini PC.

The grid is visible on Hulk's torso and feet, but it's on each of the characters (sorry not very visible on phone cam) and it moves with them.
![20230526_190006](https://github.com/flyinghead/flycast/assets/28589393/5c912d17-eaba-4b22-bf22-dfef01f844e6)
![20230526_190008](https://github.com/flyinghead/flycast/assets/28589393/6aabf358-44fa-48ce-9e55-1d270252a93f)

Changing between OGL, Vulkan, D3D9, D3D11 also didn't remove it. Any ideas?

**Comment by flyinghead** [comment, marked as answer]
If you are running at native 640x480 resolution and with default settings then this is likely a gpu or gpu driver issue. And unfortunately I don't have any workaround.
The `Native Depth Interpolation` option is recommended for AMD GPUs but I don't think it would help in this case.

**Comment by kathyrollo** [comment]
Hi @flyinghead , thanks for the quick response as always. I ran native resolution with that option enabled and the grid is still there. 3D games appear unaffected.

As much as I love these Ryzen minis, I have observed [AMD just doesn't seem to like Dreamcast emulators](https://community.amd.com/t5/drivers-software/artifacts-in-redream-emulator-regression-in-windows-gl-driver/td-p/578904). Weird quirks with their APUs and GPUs, even high-end ones. Suffice to say, Flycast did better than the other emulator. Guess I'll just have to live with this one.

**Reply by WarpedPolygon** [reply]
Just to chime in on this. After doing a whole bunch of testing on this a while back, it is always down to higher resolutions for me. I'm on AMD CPU, Nvidia GPU. Open GL 960 resolution is the max I can go without the grid lines showing on 2D games. Vulkan seems shows them no matter what. Not all 2D games suffer this issue, I personally dont go above 960 anyway so a non issue for me.

**Reply by kathyrollo** [reply]
I have no problems with my laptop which has Ryzen APU + NVIDIA GPU, where I always pick NVIDIA as my graphics adapter. I have the problem with the mini PC which is Ryzen APU only. The weird issues happen on pure AMD machines. As mentioned, the same grid appears on native resolution so it's not an upscaling issue.

UPDATE: @flyinghead It appears the option that's causing the grid on native resolution is `Forced Linear`. I set it back to `Default`. I still can't upscale beyond native but it's a workaround until AMD sorts it out.

**Comment by CO-Windler** [comment]
I am using a Ryzen 2400G APU and it plays games quite reasonable in DirectX11 mode of Win10. (Even Ferrari F355 Challenge runs fast enough.) But with OpenGL (Sega Driving Simulator crashes in other modes) some games run a bit choppy, and with Vulkan there are graphics glitches (e.g. water in the arcade game Soul Surfer). I employ a 22'' 4:3 CRT (1280*1024@72Hz) on a HDMI2VGA converter, which may look different than TFT.

**Comment by Immersion95** [comment]
Please disable "Fix upscale bleeding edge" when upscaled. You won't have the lines/grid.

**Reply by kathyrollo** [reply]
@Immersion95 I have since moved on to the PS3 HD Remaster of this game, but this tip may be useful for some. I haven't tested it, but worth a try to those having the same issues. If someone verifies this is indeed a fix, please comment to help the community. Cheers!
